docs/customizing.md

```diff
@@ -0,0 +1,238 @@
+1. Re-run `pulp_installer`.
+1. Make sure to save your variables/playbook for later usage.
+
+Example `pulp_install_plugins`:
+```
+  vars:
+    pulp_install_plugins:
+      pulp-container:
+         upgrade: true
+      pulp-file:
+         upgrade: true
+      pulp-rpm:
+         upgrade: true
+```
+
+### Specifying Exact Versions with Reproducibility:
+
+Initial installation:
+
+1. Observe the latest branch of `pulp_installer`, and what version of pulpcore it installs (3.12.2).
+1. Confirm that all the latest stable releases of your desired plugins are compatible with pulpcore
+   3.12.2, such as by reading the release announcement email thread for pulpcore 3.12.1, reading the
+plugins README, or as a last resort, reading their setup.py.
+1. If they are not all compatible yet, try the last version of the installer that installs pulpcore
+   3.11.z . Then confirm that there exist stable releases of your desired plugins that are compatible
+with pulpcore 3.11.z. If there are none, try pulpcore 3.10.z, and repeat.
+1. Once a compatible pulpcore version is found, Set `pulp_install_plugins` with each plugin listed as a key, and with each plugin having a key under it called `version` set to the version as the value (with quotes).
+1. Run `pulp_installer`
+1. Make sure to save your variables/playbook for later usage.
+
+Example `pulp_install_plugins` (with bogus version values):
+```
+  vars:
+    pulp_install_plugins:
+      pulp-container:
+         version: "4.5.6"
+      pulp-file:
+         version: "5.6.7"
+      pulp-rpm:
+         version: "6.7.8"
+```
+
+
+Upgrading your install:
+
+1. Observe what the latest version of `pulp_installer` is, and what version of pulpcore it installed
+   (3.12.2). (Even if there is no update, you can still upgrade your plugins.)
+1. Confirm that all the latest stable releases of **currently installed** plugins are compatible
+   with pulpcore 3.12.2, such as by reading the release announcement email thread for pulpcore 3.12.1,
+reading the plugins README, or as a last resort, reading their setup.py.
+1. If they are not all compatible yet, try the last version of the installer that installs pulpcore
+   3.11.z . Then confirm that there exist stable releases of your desired plugins that are compatible
+with pulpcore 3.11.z. If there are none, try pulpcore 3.10.z, and repeat.
+1. Once a compatible pulpcore version is found, Set `pulp_install_plugins` with each plugin listed as a key, and with each plugin having a key under it called `version` set to the **new** version as the value (with quotes).
+1. Upgrade `pulp_installer` to the latest version (if there is a new version.)
+1. Run `pulp_installer`
+1. Make sure to save your variables/playbook for later usage.
+
+
+Example `pulp_install_plugins` (with bogus version values):
+```
+  vars:
+    pulp_install_plugins:
+      pulp-container:
+         version: "5.6.7"
+      pulp-file:
+         version: "6.7.8"
+      pulp-rpm:
+         version: "7.8.9"
+```
```

Comment:
random thought: would it be wise to upgrade one plugin at a time?

Reply:
Honestly, that would be a safe way to go when pulpcore's minor release is unchanged.

But it would not be compatible with say upgrading from pulpcore 3.1 to 3.13, you'd need to upgrade all your plugins.